tools/sentencepiece_testing/utils.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import io
+import pathlib
+
+import sentencepiece
+
+
+def _train_sentencepiece(data, *args, **kwargs):
```

Comment:
can we just fold `_train_sentencepiece` and `_save` into one? `def train_sentencepiece(filename, data, *args, **kwargs):`

Also no need to mark this with an underscore. We only really do that for class members, and instead use our `keras_nlp_export` to mark what is public.

Reply:
Done.